### ISSUE TITLE
SimpleDataProvider in image_util

### ISSUE THREAD

**Issue by r-shruthi11** [jakeret/tf_unet#241]
I'm using the SimpleDataProvider class for my training data and my labels which are numpy arrays. I initialize a network which calls on the SimpleDataProvider but however in the training process, I see an error with the shapes of the labels array. 
The shapes of my training data are [n_samples, ny, nx, channels] and my labels are [n_samples, ny, nx, n_class] where my n_class = 2 and I use one-hot encoding to label my foreground and background pixels. 

![snip1](https://user-images.githubusercontent.com/31861615/53348270-b878bf80-38e8-11e9-883d-19c62a3e5184.PNG)
![snip2](https://user-images.githubusercontent.com/31861615/53348271-b9115600-38e8-11e9-9dea-171fd52cc4fe.PNG)

**Comment by jakeret** [comment]
I just realized that the documentation is not sufficiently accurate. 
If `n_class==2` then the implementation expects [n_samples, ny, nx, **1**] and it will transform it into a one-hot encoded tensor

**Comment by r-shruthi11** [comment]
I see, but there is no argument that the SimpleDataProvider class takes for n_class. I believe it infers it from the size of the label (label.shape[-1])

**Comment by jakeret** [comment]
Yes this is correct. [Then](https://github.com/jakeret/tf_unet/blob/master/tf_unet/image_util.py#L58) `self.n_class` will define how the labels are being processed

```
    def _process_labels(self, label):
        if self.n_class == 2:
            nx = label.shape[1]
            ny = label.shape[0]
            labels = np.zeros((ny, nx, self.n_class), dtype=np.float32)
            labels[..., 1] = label
            labels[..., 0] = ~label
            return labels
```

**Comment by r-shruthi11** [comment]
But based on your earlier comment you mention that the implementation expects [n_samples, ny, nx, **1**].  If n_class = **2** and I can't explicitly specific it then I will have to size my label array to be [n_samples, ny, nx, **2**] however this becomes incompatible with what the implementation expects.

**Comment by jakeret** [comment]
yes, you're right. Thats a bug...
In the meantime you could do something like this: 

```
data_provider = SimpleDataProvider(....)
data_provider.n_class = 2
```

**Comment by r-shruthi11** [comment]
Will do, thanks!
